Importing the necessary libraries

In [ ]:
import pandas as pd
import numpy as  np

Read my data i would work on


In [ ]:
df = pd.read_csv(r'/content/hepatitis_1.csv', sep=",")
print(df)

     age     sex steroid  antivirals  ... albumin protime histology class
0     30    male   False       False  ...     4.0     NaN     False  live
1     50  female   False       False  ...     3.5     NaN     False  live
2     78  female    True       False  ...     4.0     NaN     False  live
3     31  female     NaN        True  ...     4.0    80.0     False  live
4     34  female    True       False  ...     4.0     NaN     False  live
..   ...     ...     ...         ...  ...     ...     ...       ...   ...
150   46  female    True       False  ...     3.3    50.0      True   die
151   44  female    True       False  ...     4.3     NaN      True  live
152   61  female   False       False  ...     4.1     NaN      True  live
153   53    male   False       False  ...     4.1    48.0      True  live
154   43  female    True       False  ...     3.1    42.0      True   die

[155 rows x 20 columns]


Overview of the first 5 rows and column titles

In [ ]:
df.head(10)

age     sex steroid  antivirals  ... albumin protime histology class
0   30    male   False       False  ...     4.0     NaN     False  live
1   50  female   False       False  ...     3.5     NaN     False  live
2   78  female    True       False  ...     4.0     NaN     False  live
3   31  female     NaN        True  ...     4.0    80.0     False  live
4   34  female    True       False  ...     4.0     NaN     False  live
5   34  female    True       False  ...     4.0    75.0     False  live
6   51  female   False       False  ...     NaN     NaN     False   die
7   23  female    True       False  ...     NaN     NaN     False  live
8   39  female    True       False  ...     4.4     NaN     False  live
9   30  female    True       False  ...     3.9     NaN     False  live

[10 rows x 20 columns]

Getting a detailed information of my data set

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155 entries, 0 to 154
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   age              155 non-null    int64  
 1   sex              155 non-null    object 
 2   steroid          154 non-null    object 
 3   antivirals       155 non-null    bool   
 4   fatigue          154 non-null    object 
 5   malaise          154 non-null    object 
 6   anorexia         154 non-null    object 
 7   liver_big        145 non-null    object 
 8   liver_firm       144 non-null    object 
 9   spleen_palpable  150 non-null    object 
 10  spiders          150 non-null    object 
 11  ascites          150 non-null    object 
 12  varices          150 non-null    object 
 13  bilirubin        149 non-null    float64
 14  alk_phosphate    126 non-null    float64
 15  sgot             151 non-null    float64
 16  albumin          139 non-null    float64
 17  protime         

Overview of missing values

In [ ]:
df.isnull().sum()

age                 0
sex                 0
steroid             1
antivirals          0
fatigue             1
malaise             1
anorexia            1
liver_big          10
liver_firm         11
spleen_palpable     5
spiders             5
ascites             5
varices             5
bilirubin           6
alk_phosphate      29
sgot                4
albumin            16
protime            67
histology           0
class               0
dtype: int64

Sum total of missing values in my data set

In [ ]:
df.isnull().sum().sum()

167

Statistical description

In [ ]:
df.describe()

age   bilirubin  alk_phosphate       sgot     albumin     protime
count  155.000000  149.000000     126.000000  151.00000  139.000000   88.000000
mean    41.200000    1.427517     105.325397   85.89404    3.817266   61.852273
std     12.565878    1.212149      51.508109   89.65089    0.651523   22.875244
min      7.000000    0.300000      26.000000   14.00000    2.100000    0.000000
25%     32.000000    0.700000      74.250000   31.50000    3.400000   46.000000
50%     39.000000    1.000000      85.000000   58.00000    4.000000   61.000000
75%     50.000000    1.500000     132.250000  100.50000    4.200000   76.250000
max     78.000000    8.000000     295.000000  648.00000    6.400000  100.000000

Overview of my data set

In [ ]:
df.head(10)


age     sex steroid  antivirals  ... albumin protime histology class
0   30    male   False       False  ...     4.0     NaN     False  live
1   50  female   False       False  ...     3.5     NaN     False  live
2   78  female    True       False  ...     4.0     NaN     False  live
3   31  female     NaN        True  ...     4.0    80.0     False  live
4   34  female    True       False  ...     4.0     NaN     False  live
5   34  female    True       False  ...     4.0    75.0     False  live
6   51  female   False       False  ...     NaN     NaN     False   die
7   23  female    True       False  ...     NaN     NaN     False  live
8   39  female    True       False  ...     4.4     NaN     False  live
9   30  female    True       False  ...     3.9     NaN     False  live

[10 rows x 20 columns]

Mean values of my data set

In [ ]:
df.mean()

age                 41.200000
steroid              0.506494
antivirals           0.154839
fatigue              0.649351
malaise              0.396104
anorexia             0.207792
liver_big            0.827586
liver_firm           0.416667
spleen_palpable      0.200000
spiders              0.340000
ascites              0.133333
varices              0.120000
bilirubin            1.427517
alk_phosphate      105.325397
sgot                85.894040
albumin              3.817266
protime             61.852273
histology            0.451613
dtype: float64

Filling up missing values

In [ ]:
df1 = df.fillna(df.mean(), inplace= False)

In [ ]:
df1.head(15)

age     sex   steroid  antivirals  ...   albumin    protime histology class
0    30    male     False       False  ...  4.000000  61.852273     False  live
1    50  female     False       False  ...  3.500000  61.852273     False  live
2    78  female      True       False  ...  4.000000  61.852273     False  live
3    31  female  0.506494        True  ...  4.000000  80.000000     False  live
4    34  female      True       False  ...  4.000000  61.852273     False  live
5    34  female      True       False  ...  4.000000  75.000000     False  live
6    51  female     False       False  ...  3.817266  61.852273     False   die
7    23  female      True       False  ...  3.817266  61.852273     False  live
8    39  female      True       False  ...  4.400000  61.852273     False  live
9    30  female      True       False  ...  3.900000  61.852273     False  live
10   39  female     False        True  ...  4.400000  85.000000     False  live
11   32  female      True        True  ...  3.700000  54.000000     False  live
12   41  female      True        True  ...  3.900000  52.000000     False  live
13   30  female      True       False  ...  4.900000  78.000000     False  live
14   47  female     False        True  ...  3.817266  61.852273     False  live

[15 rows x 20 columns]

Dropping some columns and viewing the change

In [45]:
df1 = df1.drop(['steroid', 'protime'], axis=1)

In [46]:
df1.head(15)

age     sex  antivirals fatigue  ...       sgot   albumin histology class
0    30    male       False   False  ...   18.00000  4.000000     False  live
1    50  female       False    True  ...   42.00000  3.500000     False  live
2    78  female       False    True  ...   32.00000  4.000000     False  live
3    31  female        True   False  ...   52.00000  4.000000     False  live
4    34  female       False   False  ...  200.00000  4.000000     False  live
5    34  female       False   False  ...   28.00000  4.000000     False  live
6    51  female       False    True  ...   85.89404  3.817266     False   die
7    23  female       False   False  ...   85.89404  3.817266     False  live
8    39  female       False    True  ...   48.00000  4.400000     False  live
9    30  female       False   False  ...  120.00000  3.900000     False  live
10   39  female        True   False  ...   30.00000  4.400000     False  live
11   32  female        True    True  ...  249.00000  3.700000     False  live
12   41  female        True    True  ...   60.00000  3.900000     False  live
13   30  female       False    True  ...  144.00000  4.900000     False  live
14   47  female        True   False  ...   60.00000  3.817266     False  live

[15 rows x 18 columns]

Overview of the size of my data set

In [47]:
df1.shape

(155, 18)

**REPORT**

Having been given a dirty data to clean using the last class video/understanding of what has been taught.
Firstly, i downloaded my data and uploaded it on colab, then i imported the necessary libraries to be used which were pandas and numpy so as to help/ease the process.
Following that,i read the data using the csv format as it is a csv file type. I then wrote a code which displayed the details of the first five rows with all column titles.

 Next i wrote a code to get an overview of the data set i am working with which is 155 rows and 20 columns and the data type which comprises of object, bool, integer and float.

 I wrote a code next which display in summary columns with missing values and the number of missing values in each column, after that, i asked for the total of missing values which is 167. Then i wrote a code which decribed the data set (applicable to int) thus giving me a statistical background of my data set and probably give me an idea of the expected outcome i want to achieve.

Knowing the statistical part of my data set, i went ahead to get the mean of each columns with missing values which i used in filling up the missing values using the appropriate code. I decided to fill up because most columns with missing values greatly influence my anticipated outcome i wanted to achieve.

Having done that, i decide to drop 2 columns [steriod and protime] because these columns do not in any way influence my desired outcome and i could not think of anyother way to make good use of the information.

Glossary

int = integer(whole number)

bool = boolean(true/false)

float = number in decimal places

object = type of value eg name, gender etc

mean = average of a set of given values
